zqd/storage/archivestore/archivestore.go

```diff
@@ -73,8 +73,8 @@ func (s *Storage) Summary(ctx context.Context) (storage.Summary, error) {
 	return sum, err
 }
 
-func (s *Storage) Write(ctx context.Context, zctx *resolver.Context, zr zbuf.Reader) error {
-	return archive.Import(ctx, s.ark, zctx, zr)
+func (s *Storage) Write(ctx context.Context, zctx *resolver.Context, zr zbuf.Reader, maxBufSize int64) error {
+	return archive.Import(ctx, s.ark, zctx, zr, 0)
```

Comment:
I would use the package variable here and/or allow a zqd setting to adjust the value used (either with a config variable, or using the same technique we use for the sort mem max).

```suggestion
func (s *Storage) Write(ctx context.Context, zctx *resolver.Context, zr zbuf.Reader) error {
		return archive.Import(ctx, s.ark, zctx, zr, archive.DefaultImportBufSize)
```

Reply:
I get this, I didn't really like have to change the storage.Storage interface. That said my understanding from our conversation about this yesterday was that we wanted to have the ability to adjust maxBufSize on a per request basis.

> using the same technique we use for the sort mem max
The cli package does this by changing the package level var proc.MemMaxSize. We cannot really do this with zqd because of the long lived highly concurrent nature of a zqd instance.

> either with a config variable
Certainly doable if we decide to not make ImportBufSize not tunable at the request level.

If do want to make this configurable at the request level we could follow the `Options` pattern and change the storage.Storage.Write interface to `Write(context.Context, *resolver.Context, zbuf.Reader, ...storage.WriteOpts)`.

As an aside, I think we can get rid of the `*resolver.Context` as an argument in `Storage.Write`, no?

Reply:
> I get this, I didn't really like have to change the storage.Storage interface. That said my understanding from our conversation about this yesterday was that we wanted to have the ability to adjust maxBufSize on a per request basis.

Ah, I'm sorry, I should have made the distinction between two operations:
* a cli user who runs `zar import` manually: we should be able to use a cli flag to control the memory used, since the user is directly interacting with the archive & is in control of the machine running the ingest operation.
* a log import request coming over the network: the remote requester may not & should not need to know the memory characteristics of whatever instance handles their request.

So its only the cli `zar import` case that I think should be able to specify the memory size.

> > using the same technique we use for the sort mem max
> 
> The cli package does this by changing the package level var proc.MemMaxSize. We cannot really do this with zqd because of the long lived highly concurrent nature of a zqd instance.
Agreed: if we only allow the change to happen via the cli for `zar import`, like we do with zq, we're only changing the variable during the execution of that command.

> As an aside, I think we can get rid of the `*resolver.Context` as an argument in `Storage.Write`, no?
I think so!

Reply:
I'll add that I think you should be able to specify import memory size in the zqd.Config, but we can add this later.